### PR TITLE
Continued stabilization for ASB v2 audit and remediation and the CI test automation

### PR DESCRIPTION
**Pull request by MariusNi**
## Description

In this PR, the following improvements are done:

- For the SSH part of the ASB (and also going towards SSH Posture Control RTM), fixing one instance where sshd_config is written directly, replacing that with a secure write where changes are written to a temporary file, then in one atomic step renamed to the system file, preventing file corruption in case of unexpected power down, for example.
- Tuning down the error chatter that happens at the first initialization of the NRP on a machine, when the backup copy of fstab is made.
- Continuing to stabilize the automation CI for the NRP MC PS tests.

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

**Review by robertschaedler3 — APPROVED** [review]

File: .github/workflows/universalnrp-test.yml [hunk 1]
```diff
@@ -44,25 +44,4 @@ jobs:
       resource-count: ${{ matrix.policy-package.resource-count }}
       skip-remediation: ${{ matrix.skip-remediation }}
       install-osconfig: ${{ matrix.install-osconfig }}
-      tag: ${{ matrix.target.tag }}
-
-  report:
```
Comment (AhmedBM) [inline .github/workflows/universalnrp-test.yml:49]: Why is this being removed? This is what consumes the test report

Comment (AhmedBM) [reply .github/workflows/universalnrp-test.yml:49]: It is necessary, notice the hash for the commit is not updated with the latest hash
image

Comment (robertschaedler3) [reply .github/workflows/universalnrp-test.yml:49]: We should probably keep some form of a test report, odd this one is not updating to the latest commit though. I see the edits made by the bot but latest is not reflected.

**Review by robertschaedler3 — APPROVED** [review]

**Review by AhmedBM — APPROVED** [review]
👌

**Pull request merged**